backend/app/main.py

```diff
@@ -28,13 +33,66 @@
     ],
 )
 
+@app.get("/listing", response_model=ListingsGetResponseItem,
+         responses={
+            '400': {'model': ListingGetResponse},
+            '404': {'model': ListingGetResponse1},
+         })
+def get_listing(listingid: str = Query(..., description="Listing ID to retrieve")):
+    """
+    Retrieve a single listing by its listing ID
+    """
+    try:
+        # Validate listingid as a valid MongoDB ObjectId
+        if not ObjectId.is_valid(listingid):
+            raise HTTPException(status_code=400, detail="Invalid listing ID format.")
+
+        # Fetch listing from MongoDB
+        listing = db.listings.find_one({"_id": ObjectId(listingid)})
+
+        if not listing:
+            raise HTTPException(status_code=404, detail="Listing not found.")
 
-@app.get('/listings', response_model=List[ListingsGetResponseItem])
-def get_listings() -> List[ListingsGetResponseItem]:
+        # Convert MongoDB document to response model
+        return ListingsGetResponseItem(
+            id=str(listing["_id"]),
+            title=listing.get("title"),
+            price=listing.get("price"),
+            description=listing.get("description"),
+            seller_id=listing.get("user_id")  # Assuming `user_id` is the seller_id
+        )
+
+    except PyMongoError as e:
+        raise HTTPException(status_code=500, detail=f"Database error: {str(e)}")
```

Comment:
We should add a generic `500ErrorResponse` in the OpenAPI schemas and declare the endpoints that can return such an error.

Also, let's not print out stack traces in error responses since potentially exposes internal/private information. We can consider adding a debug server log for this in a future ticket.